fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugDrawer.kt

```diff
@@ -27,32 +27,26 @@ import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import org.mozilla.fenix.R
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.debugsettings.navigation.DebugDrawerDestination
 import org.mozilla.fenix.theme.FirefoxTheme
 
 /**
  * The debug drawer UI.
  *
  * @param navController [NavHostController] used to perform navigation actions on the [NavHost].
+ * @param destinations The list of [DebugDrawerDestination]s (excluding home) used to populate
+ * the [NavHost] with screens.
  * @param onBackButtonClick Invoked when the user taps on the back button in the app bar.
  */
 @Composable
 fun DebugDrawer(
     navController: NavHostController,
+    destinations: List<DebugDrawerDestination>,
     onBackButtonClick: () -> Unit,
 ) {
     var backButtonVisible by remember { mutableStateOf(false) }
     var toolbarTitle by remember { mutableStateOf("") }
 
-    // This is temporary until https://bugzilla.mozilla.org/show_bug.cgi?id=1864076
-    val homeMenuItems = List(size = 5) {
-        DebugDrawerMenuItem(
-            label = "Screen $it",
-            onClick = {
-                navController.navigate("screen_$it")
-            },
-        )
-    }
-
     Column(modifier = Modifier.fillMaxSize()) {
         TopAppBar(
```

Comment:
I did find a bug with `TopAppBar`, where the shadow is not drawing for the home screen. This will be a follow-up.